### PR TITLE
Revamp connector

### PR DESCRIPTION
**Pull request by SkNuwanTissera**
# Description

Revamp connector 
related [#14865](https://github.com/wso2-enterprise/choreo/issues/14865)

**Test Configuration**:
* Ballerina Version: Ballerina 2201.2.0
* Operating System: Ubuntu
* Java SDK: Java 11

# Checklist:

### Security checks
 - [X] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [X] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: twilio/config-types.bal [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+
+# AmazonS3 Connector configurations.
+@display {label: "Connection Config"}
+public type ConnectionConfig record {|
+    # Twilio authentication configuration
+    TokenBasedAuthentication|APIKeyBasedAuthentication auth;
+    # The HTTP version understood by the client
+    HttpVersion httpVersion = HTTP_V1_1;
+    # Configurations related to HTTP/1.x protocol
+    ClientHttp1Settings http1Settings = {};
```
Comment (LakshanSS) [inline twilio/config-types.bal:25]: Shall we add that change here?

File: twilio/config-types.bal [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+# + refreshToken - Refresh token for the token endpoint
+# + clientId - Client ID of the client authentication
+# + clientSecret - Client secret of the client authentication
+# + scopes - Scope(s) of the access request
+# + defaultTokenExpTime - Expiration time (in seconds) of the tokens if the token endpoint response does not contain an `expires_in` field
+# + clockSkew - Clock skew (in seconds) that can be used to avoid token validation failures due to clock synchronization problems
+# + optionalParams - Map of the optional parameters used for the token endpoint
+# + credentialBearer - Bearer of the authentication credentials, which is sent to the token endpoint
+# + clientConfig - HTTP client configurations, which are used to call the token endpoint
+public type OAuth2RefreshTokenGrantConfig record {|
```
Comment (LakshanSS) [inline twilio/config-types.bal:98]: As these are not used, we dont need auth related records.
Please refer to https://github.com/ballerina-platform/module-ballerinax-peoplehr/pull/13

File: twilio/Ballerina.toml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 distribution = "2201.0.3"
```
Comment (kapilraajP) [inline twilio/Ballerina.toml:2]: We can update distribution to 2201.2.0

Comment (kapilraajP) [inline twilio/Ballerina.toml]: We have to update distribution to 2201.2.0

Comment (SkNuwanTissera) [reply twilio/Ballerina.toml]: Updated

**Review by LakshanSS — APPROVED** [review]

**Pull request merged**